Allow asking for multiple instance types and subnets

github-vincent-miszczak opened the pull request.

tas50 committed:
Require Ruby 2.3 and aws-sdk-ec2

Stop depending on the monolithic AWS sdk when we only need the ec2 SDK, which is tiny in comparison.

Signed-off-by: Tim Smith <tsmith@chef.io>

Commit:
Fallback to ordered mapping when failed to detect hostname using interface_type

github-vincent-miszczak committed:
Allow asking for multiple instance types and subnets

Commit:
Add -Type parameter to userdata to support PowerShell 4.0

Signed-off-by: Stuart Preston <stuart@chef.io>

tas50 committed:
Merge pull request #425 from test-kitchen/sp/fix-ps4-ec2

Add -Type parameter to userdata to support PowerShell 4.0

Commit:
Release 2.3.1

Commit:
Merge pull request #426 from test-kitchen/rel-2.3.1

Release 2.3.1

kabakakao commented on lib/kitchen/driver/ec2.rb:
```diff
@@ -422,6 +423,53 @@ def submit_server
+        if conf[key] && conf[key].kind_of?(Array)
+          values = conf[key]
+          values.each do |value|
+            new_config = config.clone
```
This needs to be changed from config.clone to conf.clone in order to support more than one array of parameters to be expanded. We for example used both subnet & instance type together in order to have a better chance at actually getting a spot request successfully fulfilled.
Otherwise you clone the original config, which then has the key it isn't currently expanding as an array.

github-vincent-miszczak replied:
Thank you for the review, I have updated this part and making a PR upstream.

kabakakao reviewed (changes requested):
Wouldn't it be better to create a PR to the official repository? We certainly like your changes and would use them for us (we are not the maintainers of kitchen-ec2, just other users).
Also minor bug fix in a line specific comment

xtimon committed:
Add support for Windows 1709 and Windows 1803

Commit:
add amazon2

Commit:
fix image_search for amazon2

Commit:
add image selection to specs

Commit:
fix link to amis

Commit:
fix lint and wrong spec tests

Commit:
change image_search to use only hvm ami and not ecs or minimal

Commit:
fix spec

Commit:
fix amazon2 in readme file

robbkidd committed:
fix architecture comparison during image sorting

`image.architecture` returns a String, so comparing it to a symbol will always be false.

This should fix #431 for default TK+EC2 configurations returning `x86_64` architectures to the top of the preferred list of images.

tyler-ball committed:
Merge pull request #432 from test-kitchen/strings-not-symbols-yo

fix architecture comparison during image sorting

tyler-ball committed:
Adding support for arm64 architecture

Signed-off-by: tyler-ball <tball@chef.io>

robbkidd committed:
release 2.3.2

Signed-off-by: Robb Kidd <robb@thekidds.org>

tyler-ball committed:
Merge pull request #434 from test-kitchen/release-2-3-2

release 2.3.2

tyler-ball committed:
I don't see anywhere we are using winrm-fs or winrm-transport in the code

The old deprecated version of winrm-transport is causing issues updating
to the latest Test Kitchen to make tests pass.

Signed-off-by: tyler-ball <tball@chef.io>

tyler-ball committed:
Adding basic tests for StandardPlatform

Signed-off-by: tyler-ball <tball@chef.io>

tyler-ball committed:
Updating specs to work with new lifecycle_hooks key in TK 1.23

Signed-off-by: tyler-ball <tball@chef.io>

tyler-ball committed:
Merge pull request #433 from test-kitchen/arm64_support

Adding support for arm64 architecture

tyler-ball committed:
Preparing 2.3.3 release

Signed-off-by: tyler-ball <tball@chef.io>

tyler-ball committed:
Merge pull request #435 from test-kitchen/2.3.3

Preparing 2.3.3 release

tas50 committed:
Don't ship the changelog in the gem artifact

There's no need for a changelog in a runtime artifact

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge pull request #436 from test-kitchen/slim

Don't ship the changelog in the gem artifact

tas50 committed:
Release 2.3.4

Signed-off-by: Tim Smith <tsmith@chef.io>

Commit:
add changes for amazon linux 2 stable

Commit:
Merge branch 'master' into feature/amazonlinux1

Commit:
fix image selection as several tests has changed

tas50 committed:
Merge pull request #430 from pschaumburg/feature/amazonlinux1

Add support for Amazon Linux 2

Commit:
Changes for Linux as well

tas50 committed:
Merge pull request #429 from xtimon/support_windows_1709_and_1803

Add support for Windows 1709 and Windows 1803

tas50 committed:
Skip the test files in the gem

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Release 2.4.0

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Warn but properly lookup full debian versions like 9.6

If a user passes a full version like 9.6 we should warn and then lookup 9 instead instead of just failing with a bogus error.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge pull request #439 from test-kitchen/debian_versions

Warn but properly lookup full debian versions like 9.6

dpattmann committed:
Make ingress ip range configurable

tas50 committed:
Merge pull request #446 from dpattmann/security_group_cidr_ip

Make ingress ip range configurable

tas50 committed:
Merge pull request #421 from PTC-Global/fix-hostname-detection

Fallback to ordered mapping when failed to detect hostname using interface_type

tas50 committed:
Loosen the test-kitchen and yard deps

Test kitchen 2.0 has no impact on this plugin
There's no reason to constrain the yard dev dep

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge pull request #448 from test-kitchen/loosen_deps

Loosen the test-kitchen and yard deps

tas50 committed:
Remove the github changelog generator task

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Test on the latest versions of Ruby in Travis

Test on what we expect to ship on

Signed-off-by: Tim Smith <tsmith@chef.io>1

tas50 committed:
Resolve chefstyle warnings to get travis green

Fix style for the latest chefstyle

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge pull request #419 from test-kitchen/aws_v3

Require Ruby 2.3 and aws-sdk-ec2

tas50 committed:
Merge pull request #449 from test-kitchen/travis

Test on the latest versions of Ruby in Travis

tas50 committed:
Release 3.0

It's time!

Signed-off-by: Tim Smith <tsmith@chef.io>

bdwyertech committed:
Honor AWS_PROFILE

Allow AWS_PROFILE passthrough

tas50 committed:
Merge pull request #451 from bdwyertech/patch-1

Honor AWS_PROFILE

Annih committed:
Ensure tags keys are strings as expected by AWS SDK

This commit transform all tags key to Strings, because the ec2 SDK
requires it.
See https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-ec2/lib/aws-sdk-ec2/tag.rb#L260

This is a reproduction of #111 which I think reoccured when we switch
from aws-sdk to aws-sdk-ec2 in 434dc6c97d4a3c4d97ff1f1515a323dc8bc4636.

Signed-off-by: Baptiste Courtois <b.courtois@criteo.com>

Annih committed:
Leverage quadratic backoff retry on instance creation throttling

In some cases the instance creation fail due to the request limit.
Let's retry after a few seconds when it happens

Signed-off-by: Baptiste Courtois <b.courtois@criteo.com>

tas50 committed:
Merge pull request #453 from criteo-forks/retry_on_request_limit_error

Leverage quadratic backoff retry on instance creation throttling

tas50 committed:
Merge pull request #452 from criteo-forks/tags_must_be_string

Ensure tags keys are strings as expected by AWS SDK

tas50 committed:
Merge pull request #450 from test-kitchen/release_3

Release 3.0

tas50 committed:
Add missing changelog entries

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Avoid a deprecation warninig when spinning up a new instance

AWS SDK V3 doesn't allow this anymore. See https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-core/lib/aws-sdk-core/resources/collection.rb#L38

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge pull request #455 from test-kitchen/fix_deprecation

Avoid a deprecation warninig when spinning up a new instance

tas50 committed:
Release 3.0.1

Signed-off-by: Tim Smith <tsmith@chef.io>
